deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/RegisterManagerParser.java

```diff
@@ -114,10 +114,7 @@ protected boolean setDeviceId(String deviceId) throws IllegalArgumentException
         if((this.deviceId == null) || (deviceId == null) || (!this.deviceId.equals(deviceId)))
         {
             this.deviceId = deviceId;
-            if(this.deviceId != null)
-            {
-                return true;
-            }
+            return this.deviceId != null;
```

Comment:
That wouldn't work for the case where deviceId value is not null but different than this.deviceId, right? There are probably other ways to make it one line but I think it will less readable.